pallene/translator.lua

```diff
@@ -0,0 +1,81 @@
+local translator = {}
+
+local function translate_expression(expression)
+    if expression._tag == "ast.Exp.Integer" then
+        io.write(expression.value)
+    end
+end
+
+local function translate_tl_var(node)
+    io.write('local ')
+    for _, declaration in pairs(node.decls) do
+        io.write(declaration.name)
+    end
+
+    if #node.values > 0 then
+        io.write(' = ')
+        for _, value in pairs(node.values) do
+            translate_expression(value)
+        end
+    end
+
+    io.write('\n')
+end
+
+local last_index = 1
+
+local function add_previous(input, partials, stop_index)
+    local partial = input:sub(last_index, stop_index)
+    table.insert(partials, partial)
+
+    -- Do not update the last index if this is the last call to add_previous.
```

Comment:
If the last call passes in the string's length instead of `nil` then I think we wouldn't need this special case.

Reply:
Oh okay.